carto/auth.py

```diff
@@ -120,12 +134,20 @@ def send(self, relative_path, http_method, **requests_args):
 
     def prepare_send(self, http_method, **requests_args):
         http_method = http_method.lower()
+        params = {
+            "api_key": self.api_key,
+            "client": self.client_id
+        }
         if (http_method in ['post', 'put']) and "json" in requests_args:
-            requests_args["json"].update({"api_key": self.api_key})
+            requests_args["json"].update(params)
         else:
             if "params" not in requests_args:
                 requests_args["params"] = {}
-            requests_args["params"].update({"api_key": self.api_key})
+            requests_args["params"].update(params)
+
+        if 'headers' not in requests_args or not requests_args['headers']:
```

Comment:
Minor: I _think_ that you can replace this line with `if not requests_args.get('headers', None):`.